### DISCUSSION TITLE
Profile

### DISCUSSION THREAD

**Discussion by sean-regents** [relaycc/relay#95] [Ideas]
Goal: 

relay.cc/x/[address] is a link to view the profile of [address]. 

Address can be a Ethereum address [0x..], or an ENS address [seanwbren.eth], or a Lens handle [seanb.lens]. 

A user would share their own profile in order to 1) let others to learn about them and 2) invite others to chat with them. 

The content in a profile can be extensive. We will want to display: 

1. ENS Avatar
2. address + primary ENS name + lens handle
3. web2 handles / links
4. User bio
5. NFT trophy case (5-10 NFTs selected as favorites / representative of the person, see dome inspiration)
6. Web3 profile cards. for example, an opensea profile card would show their opensea profile picture and link out to opensea profile.

Inspiration: 
https://www.mydome.xyz/sean
https://light.so/seanwbren
https://welook.io/seanwbren.eth

Design:
3D cards: https://garden.bradwoods.io/notes/css/3d#usecasedifferentperspective 
3D cards: https://github.com/simeydotme/pokemon-cards-css --> cool demo!

Further questions: 
1. This mainly describes a user profile. What changes would a company profile or smart contract profile have? See alchemy & coinbase directory for company profile inspo.

**Comment by mmmarcusb** [comment]
Imagining there would also be a view split between "Someone else's profile" and "Your own profile"

"Someone else's profile" would have buttons for adding as a Relay contact, and messaging them.

Additionally, the "mini-profile" view in Receiver etc. would display something like 1, 2, 3, 4, and maybe the top NFT from 5?  Or all of them.  I'm unsure of the real estate on this one.

**Reply by sean-regents** [reply]
Receiver profiles should have a separate discussion topic, which would be a subset of the features in here. Receiver will have your own profile view + profile previews, while relay.cc will have the same but more screen real estate to work with

**Comment by sean-regents** [comment]
3D Design discussion 1/12
Sean: 
specifically, what if parts of the profile or cards had the animations in this: https://poke-holo.simey.me
the first time a card is clicked, it has the spin effect. Every other time it just grows to fill more of the view
App cards / Badges / NFTs having that grow effect plus more details appearing on the card could be very useful
It's TBD on design, though if it's useful + fun for directory/profiles than we should try it

BrianS:
They are easily buildable. It depends on what user experience we want to provide to our users
That isnt even a 3d environment. That's pure css. Skewing the object on mouse position and implementing gradients
App cards / Badges / NFTs having that grow effect plus more details appearing on the card could be very useful
If that is the design opinion we come to I can do it for sure
